src/Microsoft.Data.SqlClient/netfx/src/Microsoft/Data/SqlClient/EnclaveSessionCache.cs

```diff
@@ -67,9 +67,9 @@ internal SqlEnclaveSession CreateSession(string attestationUrl, string serverNam
         }
 
         // Generates the cache key for the enclave session cache
-        private string GenerateCacheKey(string serverName, string attestationUrl)
+        private string GenerateCacheKey(string serverName, string attestationUrl, string database)
         {
-            return (serverName + attestationUrl).ToLowerInvariant();
+            return (serverName + database + attestationUrl).ToLowerInvariant();
```

Comment:
It would be good to add a delimiter character that is not possible to be in the database/server identifier. For example:
If you have a server named `AAB` and a database named `BCC` then together you get `AABBCC`.
Likewise, if you have a server named `AABB` and a database named CC then you still get `AABBCC`
But with a delimiter like '+' then the concatenations would be different: `AAB+BCC` vs `AABB+CC`